lib/preserve/callback.rb

```diff
@@ -65,5 +67,10 @@ def parameter_value=(value)
     def session_key
       SessionKey.new(source_class, parameter_key).build
     end
+
+    def parameter_to_hash(params)
+      params.permit! if params.respond_to?(:permit!)
+      params.to_hash
+    end
```

Comment:
I think we need to clone the method argument here, otherwise calling `permit!` would mutate the `params` object in the controller. That would have an undesired side-effect of permitting all nested parameters.
```suggestion
    def parameters_to_hash(value)
      params = value.dup
      params.permit! if params.respond_to?(:permit!)
      params.to_hash
    end
```

Reply:
You're right. But I found an easier way. 

See https://api.rubyonrails.org/classes/ActionController/Parameters.html#method-i-to_unsafe_h

Let me know. Thanks for the review

Reply:
That works too (as long as it doesn't change the `params` object) 👍